diff --git a/dist/compiled/@vercel/og/index.edge.js b/dist/compiled/@vercel/og/index.edge.js
index de7cd378a4f1f2d07def4714b9c440352709e699..ce63f7ce7969e9589d4501eceeeb8ef411b0fa68 100644
--- a/dist/compiled/@vercel/og/index.edge.js
+++ b/dist/compiled/@vercel/og/index.edge.js
@@ -17935,7 +17935,7 @@ async function* Ki(e, t) {
   let xe = s.getEngine(P2, S2, r, h2), Je = m2 ? Ve(se, "grapheme").filter((Y) => !Ad(Y) && !xe.has(Y)) : [];
   yield Je.map((Y) => ({ word: Y, locale: h2 })), Je.length && (xe = s.getEngine(P2, S2, r, h2));
   function Pe(Y) {
-    return !!(p && p[Y]);
+    return !!(p && Object.prototype.hasOwnProperty.call(p, Y) && p[Y]);
   }
   let { measureGrapheme: be, measureGraphemeArray: It, measureText: Ae } = tu(xe, Pe, { fontSize: P2, letterSpacing: O }), ae = Bn(k) ? X(k, P2, 1, r) : be(nr) * k, H = (Y, V) => {
     if (Y.length === 0)
@@ -18018,7 +18018,7 @@ async function* Ki(e, t) {
     let V = De[Y], ue = De[Y + 1];
     if (!V)
       continue;
-    let K2 = Se[Y], ve = null, oe = false, _e = p ? p[K2] : null, ee = V.y, te = V.x, ot = V.width, Q = V.line, Fe = r.textDecorationLine === "underline" && (r.textDecorationSkipInk || "auto") !== "none";
+    let K2 = Se[Y], ve = null, oe = false, _e = p && Object.prototype.hasOwnProperty.call(p, K2) ? p[K2] : null, ee = V.y, te = V.x, ot = V.width, Q = V.line, Fe = r.textDecorationLine === "underline" && (r.textDecorationSkipInk || "auto") !== "none";
     if (Q === qt)
       continue;
     let ze = false;
diff --git a/dist/compiled/@vercel/og/index.node.js b/dist/compiled/@vercel/og/index.node.js
index 127a29b52a0e898c229083894de406fe76bab55c..f106d037cc75b9d69c9f73795a4efdc638cc8737 100644
--- a/dist/compiled/@vercel/og/index.node.js
+++ b/dist/compiled/@vercel/og/index.node.js
@@ -18937,7 +18937,7 @@ async function* li(A, e) {
   let uA = o.getEngine(M, v2, r, C), te = d2 ? ge(oA, "grapheme").filter((z) => !Ul(z) && !uA.has(z)) : [];
   yield te.map((z) => ({ word: z, locale: C })), te.length && (uA = o.getEngine(M, v2, r, C));
   function xA(z) {
-    return !!(E && E[z]);
+    return !!(E && Object.prototype.hasOwnProperty.call(E, z) && E[z]);
   }
   let { measureGrapheme: yA, measureGraphemeArray: me, measureText: RA } = Ia(uA, xA, { fontSize: M, letterSpacing: F }), fA = Xr(k) ? tA(k, M, 1, r) : yA(Qt) * k, q = (z, V) => {
     if (z.length === 0)
@@ -19020,7 +19020,7 @@ async function* li(A, e) {
     let V = OA[z], iA = OA[z + 1];
     if (!V)
       continue;
-    let $ = K2[z], IA = null, lA = false, SA = E ? E[$] : null, cA = V.y, aA = V.x, Be = V.width, eA = V.line, PA = r.textDecorationLine === "underline" && (r.textDecorationSkipInk || "auto") !== "none";
+    let $ = K2[z], IA = null, lA = false, SA = E && Object.prototype.hasOwnProperty.call(E, $) ? E[$] : null, cA = V.y, aA = V.x, Be = V.width, eA = V.line, PA = r.textDecorationLine === "underline" && (r.textDecorationSkipInk || "auto") !== "none";
     if (eA === De)
       continue;
     let NA = false;
